src/views/Root/Root.js

```diff
@@ -0,0 +1,141 @@
+import React, { Component } from 'react';
+import './index.css';
+import { BrowserRouter, Route, Switch } from 'react-router-dom';
+import AppContext from '../../context';
+import Modal from '../../components/Modal/Modal';
+import Header from '../../components/Header/Header';
+import GroceryListView from '../GroceryListView/GroceryListView';
+import RecipesListView from '../RecipesListView/RecipesListView';
+import StatisticsView from '../StatisticsView/StatisticsView';
+import Form from '../../components/Form/Form';
+import FormTypes from '../../components/Form/FormTypes';
+import { GroceryService } from './GroceryService';
+
+class Root extends Component {
+	state = {
+		isModalOpen: false,
+		formType: FormTypes.addItem,
+		grocery: [],
+		searchQuery: '',
+		groceryCompleted: [],
+		itemUnderEdition: '',
+		isNameSearch: true,
+		showCompleted: true
+	};
+
+	groceryService = new GroceryService();
+
+	componentWillMount = () => {
+		this.setState({
+			grocery: this.groceryService.getAllGroceries(),
+			groceryCompleted: this.groceryService.getAllCompletedGroceries()
+		});
+	};
+
+	closeModal = () => {
+		this.setState({
+			isModalOpen: false,
+			itemUnderEdition: null
+		});
+	};
+
+	openAddItemModal = () => {
+		this.setState({
+			isModalOpen: true,
+			formType: FormTypes.addItem
+		});
+	};
+
+	addItem = (event, newItem) => {
+		event.preventDefault();
+		const grocery = this.groceryService.addGroceryItem(newItem);
+
+		this.setState({
+			grocery: grocery,
+		});
+
+		this.closeModal();
+	};
+
+	deleteItem = (name) => {
+		const grocery = this.groceryService.deleteGroceryItem(name);
+
+		this.setState({
+			grocery: grocery
+		});
+	};
+
+	updateItem = (event, item) => {
+		event.preventDefault();
+		const updateResult = this.groceryService.updateGroceryItem(item);
+
+		this.setState({
+			grocery: updateResult.grocery,
+			groceryCompleted: updateResult.groceryCompleted,
+		});
+
+		this.closeModal();
+	};
+
+	handleSearchInput = event => {
+		const searchQuery = event.target.value;
+		this.setState({
+			searchQuery: searchQuery
+		});
+	};
+
+	markAsCompleted = (name) => {
+		this.setState({
+			isModalOpen: true,
+			formType: FormTypes.addPrice,
+			itemUnderEdition: name
+		});
+	};
+
+	switchSearch = () => {
+		this.setState(prevState => ({
+			isNameSearch: !prevState.isNameSearch
+		}));
+	};
+
+	toggleShowCompleted = () => {
+		this.setState(prevState => ({
+			showCompleted: !prevState.showCompleted
+		}));
+	};
+
+	render() {
+		const { isModalOpen, formType, itemUnderEdition } = this.state;
+		const contextElements = {
```

Comment:
It's cool you considered you don't need Redux, but instead of having a huge `AppContext` have smaller context that would manage each pice of state separately. E.g: `ItemsContext`, `UIContext`, etc... and components would use the one they specifically need.